File: spring-skeleton/frontend/views/vista-vertweetgeneral.ts
import { LitElement, html, css, customElement } from 'lit-element';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import '@vaadin/horizontal-layout/src/vaadin-horizontal-layout.js';
import '@vaadin/vertical-layout/src/vaadin-vertical-layout.js';
import '@vaadin/button/src/vaadin-button.js';

@customElement('vista-vertweetgeneral')
export class VistaVertweetgeneral extends LitElement {
  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
<vaadin-vertical-layout id="verticalLayout-VentanaCompleta">
 <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; width: 88%; background-color: #cdc8c8; align-self: center;" id="horizontalLayout-tweetCompleto">
  <vaadin-vertical-layout theme="spacing" style="flex-grow: 1;" id="verticalLayout-TweetCompleto">
   <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; width: 100%; height: 15%;" id="horizontalLayout-Cabecera">
    <img style="flex-grow: 1; width: 38%; height: 100%; margin-top: var(--lumo-space-s); margin-left: var(--lumo-space-s);" id="img-FotoPerfilTweet">
    <label style="flex-grow: 1; width: 100%; margin-top: var(--lumo-space-xl);" id="label-NombreUsuario">Nombre Usuario</label>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; height: 70%; width: 100%;" id="vaadinHorizontalLayout1">
    <vaadin-vertical-layout theme="spacing" style="flex-grow: 1;" id="vaadinVerticalLayout2">
     <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; width: 100%;" id="horizontalLayout-CuerpoTweet">
      <label style="flex-grow: 1; margin-top: var(--lumo-space-m); margin-left: var(--lumo-space-m);" id="label-CuerpoTwet">Texto tweet</label>
     </vaadin-horizontal-layout>
     <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; width: 100%;" id="horizontalLayout-Multimedia">
      <vaadin-vertical-layout theme="spacing" style="flex-grow: 1;" id="verticalLayout-Multimedia">
       <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; width: 60%; align-self: center;" id="horizontal-Multimedia">
        <img style="flex-grow: 1;" id="img-FotoTweet">
        <div style="flex-grow: 1;" id="div-Tweet">
          Div 
        </div>
       </vaadin-horizontal-layout>
      </vaadin-vertical-layout>
     </vaadin-horizontal-layout>
    </vaadin-vertical-layout>
   </vaadin-horizontal-layout>
   <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; width: 100%;" id="horizontalLayout-Opciones">
    <vaadin-vertical-layout theme="spacing" style="flex-grow: 1; width: 20%;" id="verticalLayout-RetweetBorrar">
     <vaadin-button style="flex-grow: 1; align-self: flex-start; width: 30%; margin-left: var(--lumo-space-m);" tabindex="0" id="vaadinButton">
       Comentar 
     </vaadin-button>
     <vaadin-button style="flex-grow: 1; align-self: flex-start; width: 30%; margin-left: var(--lumo-space-m);" tabindex="0" id="button-Retweet">
       Retweetear 
     </vaadin-button>
     <vaadin-button style="flex-grow: 1; align-self: flex-start; width: 30%; margin-left: var(--lumo-space-m);" tabindex="0" id="button-Borrar">
       Borrar 
     </vaadin-button>
    </vaadin-vertical-layout>
    <vaadin-vertical-layout theme="spacing" style="flex-grow: 1;" id="verticalLayout-EspacioBotones"></vaadin-vertical-layout>
    <vaadin-vertical-layout theme="spacing" style="flex-grow: 1; width: 20%; align-self: flex-end; margin-right: var(--lumo-space-m); margin-bottom: var(--lumo-space-xl);" id="verticalLayout-MeGusta">
     <vaadin-horizontal-layout theme="spacing" id="horizontalLayout-MeGusta" style="align-self: flex-end; margin-right: var(--lumo-space-s);">
      <label style="flex-grow: 1;" id="label-MeGusta">Num Me gusta</label>
      <img style="flex-grow: 1;" id="img-MeGusta">
      <vaadin-button theme="icon" aria-label="Add new" style="flex-grow: 1;" tabindex="0" id="button-MeGusta">
       <vaadin-button id="icon-MeGusta" tabindex="0">
         Me gusta 
       </vaadin-button>
      </vaadin-button>
     </vaadin-horizontal-layout>
    </vaadin-vertical-layout>
   </vaadin-horizontal-layout>
  </vaadin-vertical-layout>
 </vaadin-horizontal-layout>
 <vaadin-vertical-layout theme="spacing" style="align-self: center; flex-grow: 1; width: 60%;" id="vaadinVerticalLayout"></vaadin-vertical-layout>
</vaadin-vertical-layout>
`;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}

File: spring-skeleton/src/main/java/vistas/VistaVertweetgeneral.java
package vistas;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.dom.Element;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.button.Button;

@Tag("vista-vertweetgeneral")
@JsModule("./views/vista-vertweetgeneral.ts")
public class VistaVertweetgeneral extends LitTemplate {

    @Id("verticalLayout-VentanaCompleta")
	private Element verticalLayoutVentanaCompleta;
	@Id("horizontalLayout-tweetCompleto")
	private HorizontalLayout horizontalLayoutTweetCompleto;
	@Id("verticalLayout-TweetCompleto")
	private Element verticalLayoutTweetCompleto;
	@Id("horizontalLayout-Cabecera")
	private HorizontalLayout horizontalLayoutCabecera;
	@Id("img-FotoPerfilTweet")
	private Image imgFotoPerfilTweet;
	@Id("label-NombreUsuario")
	private Label labelNombreUsuario;
	@Id("vaadinHorizontalLayout1")
	private HorizontalLayout vaadinHorizontalLayout1;
	@Id("vaadinVerticalLayout2")
	private Element vaadinVerticalLayout2;
	@Id("horizontalLayout-CuerpoTweet")
	private HorizontalLayout horizontalLayoutCuerpoTweet;
	@Id("label-CuerpoTwet")
	private Label labelCuerpoTwet;
	@Id("horizontalLayout-Multimedia")
	private HorizontalLayout horizontalLayoutMultimedia;
	@Id("verticalLayout-Multimedia")
	private Element verticalLayoutMultimedia;
	@Id("horizontal-Multimedia")
	private HorizontalLayout horizontalMultimedia;
	@Id("img-FotoTweet")
	private Image imgFotoTweet;
	@Id("div-Tweet")
	private Div divTweet;
	@Id("horizontalLayout-Opciones")
	private HorizontalLayout horizontalLayoutOpciones;
	@Id("verticalLayout-RetweetBorrar")
	private Element verticalLayoutRetweetBorrar;
	@Id("button-Retweet")
	private Button buttonRetweet;
	@Id("button-Borrar")
	private Button buttonBorrar;
	@Id("verticalLayout-EspacioBotones")
	private Element verticalLayoutEspacioBotones;
	@Id("verticalLayout-MeGusta")
	private Element verticalLayoutMeGusta;
	@Id("horizontalLayout-MeGusta")
	private HorizontalLayout horizontalLayoutMeGusta;
	@Id("label-MeGusta")
	private Label labelMeGusta;
	@Id("img-MeGusta")
	private Image imgMeGusta;
	@Id("button-MeGusta")
	private Button buttonMeGusta;
	@Id("icon-MeGusta")
	private Button iconMeGusta;
	@Id("vaadinVerticalLayout")
	private Element vaadinVerticalLayout;
	@Id("button-comentar")
	private Button buttonComentar;
	@Id("button-Atras")
	private Button buttonAtras;

	public Button getButtonAtras() {
		return buttonAtras;
	}

	public void setButtonAtras(Button buttonAtras) {
		this.buttonAtras = buttonAtras;
	}

	public VistaVertweetgeneral() {
        // You can initialise any data required for the connected UI components here.
    }

	public Element getVerticalLayoutVentanaCompleta() {
		return verticalLayoutVentanaCompleta;
	}

	public void setVerticalLayoutVentanaCompleta(Element verticalLayoutVentanaCompleta) {
		this.verticalLayoutVentanaCompleta = verticalLayoutVentanaCompleta;
	}

	public HorizontalLayout getHorizontalLayoutTweetCompleto() {
		return horizontalLayoutTweetCompleto;
	}

	public void setHorizontalLayoutTweetCompleto(HorizontalLayout horizontalLayoutTweetCompleto) {
		this.horizontalLayoutTweetCompleto = horizontalLayoutTweetCompleto;
	}

	public Element getVerticalLayoutTweetCompleto() {
		return verticalLayoutTweetCompleto;
	}

	public void setVerticalLayoutTweetCompleto(Element verticalLayoutTweetCompleto) {
		this.verticalLayoutTweetCompleto = verticalLayoutTweetCompleto;
	}

	public HorizontalLayout getHorizontalLayoutCabecera() {
		return horizontalLayoutCabecera;
	}

	public void setHorizontalLayoutCabecera(HorizontalLayout horizontalLayoutCabecera) {
		this.horizontalLayoutCabecera = horizontalLayoutCabecera;
	}

	public Image getImgFotoPerfilTweet() {
		return imgFotoPerfilTweet;
	}

	public void setImgFotoPerfilTweet(Image imgFotoPerfilTweet) {
		this.imgFotoPerfilTweet = imgFotoPerfilTweet;
	}

	public Label getLabelNombreUsuario() {
		return labelNombreUsuario;
	}

	public void setLabelNombreUsuario(Label labelNombreUsuario) {
		this.labelNombreUsuario = labelNombreUsuario;
	}

	public HorizontalLayout getVaadinHorizontalLayout1() {
		return vaadinHorizontalLayout1;
	}

	public void setVaadinHorizontalLayout1(HorizontalLayout vaadinHorizontalLayout1) {
		this.vaadinHorizontalLayout1 = vaadinHorizontalLayout1;
	}

	public Element getVaadinVerticalLayout2() {
		return vaadinVerticalLayout2;
	}

	public void setVaadinVerticalLayout2(Element vaadinVerticalLayout2) {
		this.vaadinVerticalLayout2 = vaadinVerticalLayout2;
	}

	public HorizontalLayout getHorizontalLayoutCuerpoTweet() {
		return horizontalLayoutCuerpoTweet;
	}

	public void setHorizontalLayoutCuerpoTweet(HorizontalLayout horizontalLayoutCuerpoTweet) {
		this.horizontalLayoutCuerpoTweet = horizontalLayoutCuerpoTweet;
	}

	public Label getLabelCuerpoTwet() {
		return labelCuerpoTwet;
	}

	public void setLabelCuerpoTwet(Label labelCuerpoTwet) {
		this.labelCuerpoTwet = labelCuerpoTwet;
	}

	public HorizontalLayout getHorizontalLayoutMultimedia() {
		return horizontalLayoutMultimedia;
	}

	public void setHorizontalLayoutMultimedia(HorizontalLayout horizontalLayoutMultimedia) {
		this.horizontalLayoutMultimedia = horizontalLayoutMultimedia;
	}

	public Element getVerticalLayoutMultimedia() {
		return verticalLayoutMultimedia;
	}

	public void setVerticalLayoutMultimedia(Element verticalLayoutMultimedia) {
		this.verticalLayoutMultimedia = verticalLayoutMultimedia;
	}

	public HorizontalLayout getHorizontalMultimedia() {
		return horizontalMultimedia;
	}

	public void setHorizontalMultimedia(HorizontalLayout horizontalMultimedia) {
		this.horizontalMultimedia = horizontalMultimedia;
	}

	public Image getImgFotoTweet() {
		return imgFotoTweet;
	}

	public void setImgFotoTweet(Image imgFotoTweet) {
		this.imgFotoTweet = imgFotoTweet;
	}

	public Div getDivTweet() {
		return divTweet;
	}

	public void setDivTweet(Div divTweet) {
		this.divTweet = divTweet;
	}

	public HorizontalLayout getHorizontalLayoutOpciones() {
		return horizontalLayoutOpciones;
	}

	public void setHorizontalLayoutOpciones(HorizontalLayout horizontalLayoutOpciones) {
		this.horizontalLayoutOpciones = horizontalLayoutOpciones;
	}

	public Element getVerticalLayoutRetweetBorrar() {
		return verticalLayoutRetweetBorrar;
	}

	public void setVerticalLayoutRetweetBorrar(Element verticalLayoutRetweetBorrar) {
		this.verticalLayoutRetweetBorrar = verticalLayoutRetweetBorrar;
	}

	public Button getButtonRetweet() {
		return buttonRetweet;
	}

	public void setButtonRetweet(Button buttonRetweet) {
		this.buttonRetweet = buttonRetweet;
	}

	public Button getButtonBorrar() {
		return buttonBorrar;
	}

	public void setButtonBorrar(Button buttonBorrar) {
		this.buttonBorrar = buttonBorrar;
	}

	public Element getVerticalLayoutEspacioBotones() {
		return verticalLayoutEspacioBotones;
	}

	public void setVerticalLayoutEspacioBotones(Element verticalLayoutEspacioBotones) {
		this.verticalLayoutEspacioBotones = verticalLayoutEspacioBotones;
	}

	public Element getVerticalLayoutMeGusta() {
		return verticalLayoutMeGusta;
	}

	public void setVerticalLayoutMeGusta(Element verticalLayoutMeGusta) {
		this.verticalLayoutMeGusta = verticalLayoutMeGusta;
	}

	public HorizontalLayout getHorizontalLayoutMeGusta() {
		return horizontalLayoutMeGusta;
	}

	public void setHorizontalLayoutMeGusta(HorizontalLayout horizontalLayoutMeGusta) {
		this.horizontalLayoutMeGusta = horizontalLayoutMeGusta;
	}

	public Label getLabelMeGusta() {
		return labelMeGusta;
	}

	public void setLabelMeGusta(Label labelMeGusta) {
		this.labelMeGusta = labelMeGusta;
	}

	public Image getImgMeGusta() {
		return imgMeGusta;
	}

	public void setImgMeGusta(Image imgMeGusta) {
		this.imgMeGusta = imgMeGusta;
	}

	public Button getButtonMeGusta() {
		return buttonMeGusta;
	}

	public void setButtonMeGusta(Button buttonMeGusta) {
		this.buttonMeGusta = buttonMeGusta;
	}

	public Button getIconMeGusta() {
		return iconMeGusta;
	}

	public void setIconMeGusta(Button iconMeGusta) {
		this.iconMeGusta = iconMeGusta;
	}

	public Element getVaadinVerticalLayout() {
		return vaadinVerticalLayout;
	}

	public void setVaadinVerticalLayout(Element vaadinVerticalLayout) {
		this.vaadinVerticalLayout = vaadinVerticalLayout;
	}
	
	public Button getButtonComentar() {
		return buttonComentar;
	}

	public void setButtonComentar(Button buttonComentar) {
		this.buttonComentar = buttonComentar;
	}
	
    
    
    
    
    
    
    
    

}